HW06.Rmd

```diff
@@ -1,92 +1,115 @@
+                         keytype = "TXID", columns = c("SYMBOL", "TXID"))
+
+ens.mapped_GBM <- left_join(ens.GBM.transcripts, map, by = c("gene_id" = "TXID")) %>% 
+  dplyr::select(-gene_id) %>%     # Maybe gene_id is not the first column
+  dplyr::select(gene_symbol = SYMBOL, everything())
+
+ens.mapped_GBM <- ens.mapped_GBM[!duplicated(ens.mapped_GBM$gene_symbol),] #remove duplicated gene symbols
+  #these are removed instead of averaged because they simply do not correlate particularly well. 
+ens.mapped_GBM <- ens.mapped_GBM[!is.na(ens.mapped_GBM$gene_symbol),] #remove NA values
+
+
+```
+
+
+
+Do the t-test and make a table of the t-test results!
+
+```{r t.test for difference between a bunch of genes}
+#run this code to unload the libraries from before, it might be helpful because the select() function from dplyr might be hidden otherwise
+detach(package:EnsDb.Hsapiens.v75, unload = T)
+detach(package:ensembldb, unload = T)
+
+#add in your own gene of interest!!! 
+genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2")
+
+# Do a bit more: transpose, and separate whether a cell line is WT or mutant
+GBM.genes.of.interest <- ens.mapped_GBM %>%
+  filter(gene_symbol %in% genes_of_interest) %>%
+  pivot_longer(-gene_symbol, names_to = "cell_line", values_to = "TPM") %>%     # tidy up!
+  separate(cell_line, c("mutant", "cell_line"), sep = '/', convert = TRUE)     # the name of the parent folder was reserved for `mutant` column
+
+#Now perform a t-test between the H3K mutated and the wt samples. There are many ways to do this actually, you can use a for loop or you could do the tidy alternative with broom(), but the for loop is probably the easiest
+GBM.genes.of.interest %>%
+  group_by(gene_symbol) %>%
+  do(tidy(t.test(filter(., mutant == "WT")$TPM, filter(., mutant == "H3K27M")$TPM))) %>%
+  kable() #print out the t-test results
+
+```
+
+
+
+Now create a graphing function to create boxplots to visualize the results. Plot expression on the y-axis. The graph should look like [this example](IRX1_example.png)
+
+```{r create a graphing function}
+#create a graphing function
+my_boxplot <- function(gene, data) {
+  data %>% filter(gene_symbol == gene) %>%
+    ggplot(aes(mutant, TPM)) +
+      geom_boxplot(aes(fill = mutant)) +
+      labs(title = paste(gene, "Expression in GBM models by H3K27 Mutated or WT Status"), y = "Expression(TPM)")
+}
+
+#then use a for loop combined with the graphing function to make a graph for all your genes of interest 
+lapply(unique(GBM.genes.of.interest$gene_symbol), my_boxplot, data = GBM.genes.of.interest)
+```
+
+In the end of the day I was able to satisfy the prompts by adding exactly **20** lines of codes, not counting library inclusion, empty lines and comment lines.
```

Comment:
Very impressive! I am working on this as well, but I think a bit more commenting within the code might be helpful for some areas, such as the data call section.